app/api/services/suppliers.py

```diff
@@ -172,15 +178,59 @@ def get_suppliers_with_expiring_documents(self, days):
                                               func.json_agg(
                                                   email_addresses.columns.email_address
                                               ).label('email_addresses'))
-                                       .group_by(email_addresses.columns.code)
-                                       .subquery())
+                             .group_by(email_addresses.columns.code)
+                             .subquery())
 
         # Combine the list of email addresses and documents
         results = (db.session.query(documents.columns.code, documents.columns.name, documents.columns.documents,
                                     aggregated_emails.columns.email_addresses)
-                             .join(aggregated_emails,
-                                   documents.columns.code == aggregated_emails.columns.code)
-                             .order_by(documents.columns.code)
-                             .all())
+                   .join(aggregated_emails,
+                         documents.columns.code == aggregated_emails.columns.code)
+                   .order_by(documents.columns.code)
+                   .all())
+
+        return [r._asdict() for r in results]
+
+    def get_suppliers_with_unassessed_domains_and_all_case_studies_rejected(self):
+        case_study_query = (
+            db.session.query(
+                CaseStudy.supplier_code.label('supplier_code'),
+                CaseStudy.data['service'].astext.label('domain'),
+                func.count(CaseStudy.id).label('count')
+            )
+            .group_by(CaseStudy.supplier_code, CaseStudy.data['service'].astext)
+        )
+
+        subquery = (
+            case_study_query
+            .intersect(
+                case_study_query
+                .filter(CaseStudy.status == 'rejected')
+            )
+            .subquery()
+        )
+
+        results = (
+            db
+            .session
+            .query(
+                Supplier.id,
+                Supplier.code,
+                Supplier.name,
+                func.json_agg(aggregate_order_by(Domain.name, Domain.name)).label('domains')
+            )
+            .join(SupplierDomain, Domain)
+            .join(subquery, and_(
+                Supplier.code == subquery.columns.supplier_code,
+                Domain.name == subquery.columns.domain
+            ))
+            .filter(
+                Supplier.status != 'deleted',
+                Supplier.data['recruiter'].astext.in_(['no', 'both']),
```

Comment:
How do we handle the case when no value exists for `recruiter`?

Reply:
I don't believe this is a concern for this code, because it would mean we would have to update all queries to take missing recruiter into account.  The data should be fixed instead.